app/screens/channel_files/channel_files.tsx

```diff
@@ -0,0 +1,169 @@
+
+import Header from './header';
+
+import type ChannelModel from '@typings/database/models/servers/channel';
+
+const TEST_ID = 'channel_files';
+
+type Props = {
+    channel: ChannelModel;
+    canDownloadFiles: boolean;
+    publicLinkEnabled: boolean;
+}
+
+const edges: Edge[] = ['bottom', 'left', 'right'];
+
+const styles = StyleSheet.create({
+    flex: {
+        flex: 1,
+    },
+    empty: {
+        alignItems: 'center',
+        flex: 1,
+        justifyContent: 'center',
+    },
+    list: {
+        paddingVertical: 8,
+    },
+    loading: {
+        justifyContent: 'center',
+        flex: 1,
+    },
+    searchBar: {
+        marginLeft: 20,
+        marginRight: Platform.select({ios: 12, default: 20}),
+        marginTop: 20,
+    },
+});
+
+const getSearchParams = (channelId: string, searchTerm?: string, filterValue?: FileFilter) => {
+    const term = `channel:${channelId}`;
+    const fileExtensions = filterFileExtensions(filterValue);
+    const extensionTerms = fileExtensions ? ' ' + fileExtensions : '';
+    const searchTerms = searchTerm ? ' ' + searchTerm : '';
+    return {
+        terms: term + searchTerms + extensionTerms,
+        is_or_search: true,
+    };
+};
+
+const emptyFileResults: FileInfo[] = [];
+
+function ChannelFiles({
+    channel,
+    canDownloadFiles,
+    publicLinkEnabled,
+}: Props) {
+    const theme = useTheme();
+    const serverUrl = useServerUrl();
+    const {formatMessage} = useIntl();
+    const searchTimeoutId = useRef<NodeJS.Timeout | null>(null);
+    const [filter, setFilter] = useState<FileFilter>(FileFilters.ALL);
+    const [loading, setLoading] = useState(true);
+    const [term, setTerm] = useState('');
+    const lastSearchRequest = useRef<number>();
+
+    const [fileInfos, setFileInfos] = useState<FileInfo[]>(emptyFileResults);
+
+    const handleSearch = useCallback(async (searchTerm: string, ftr: FileFilter) => {
+        const t = Date.now();
+        lastSearchRequest.current = t;
+        const searchParams = getSearchParams(channel.id, searchTerm, ftr);
+        const {files} = await searchFiles(serverUrl, channel.teamId, searchParams);
+        if (lastSearchRequest.current !== t) {
+            return;
+        }
+        setFileInfos(files?.length ? files : emptyFileResults);
+        setLoading(false);
+    }, [serverUrl, channel]);
+
+    useEffect(() => {
+        if (searchTimeoutId.current) {
+            clearTimeout(searchTimeoutId.current);
+        }
+
+        searchTimeoutId.current = setTimeout(() => {
+            handleSearch(term, filter);
+        }, General.SEARCH_TIMEOUT_MILLISECONDS);
+    }, [filter, term, handleSearch]);
+
+    const handleFilterChange = useCallback(async (filterValue: FileFilter) => {
+        setLoading(true);
+        setFilter(filterValue);
+    }, []);
+
+    const clearSearch = useCallback(() => {
+        setTerm('');
+    }, []);
+
+    const onTextChange = useCallback((searchTerm: string) => {
```

Comment:
I think we want to debounce this so that is not performed on every key press, it looks like too much at the moment, also there is no request sequence control, so if one request finishes before the other, results may not match the search

Reply:
nvm I see that you have a use effect with a timeout for this